packages/rum-core/src/performance-monitoring/perf-entry-recorder.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { LONG_TASK, LARGEST_CONTENTFUL_PAINT } from '../common/constants'
+import { noop } from '../common/utils'
+import Span from './span'
+
+/**
+ * Create Spans for the long task entries
+ * Spec - https://w3c.github.io/longtasks/
+ */
+function createLongTaskSpans(longtasks) {
+  const spans = []
+
+  for (let i = 0; i < longtasks.length; i++) {
+    /**
+     * name of the long task can be any one of the type listed here
+     * depeding on the origin of the longtask
+     * https://w3c.github.io/longtasks/#sec-PerformanceLongTaskTiming
+     */
+    const { name, startTime, duration, attribution } = longtasks[i]
+    const end = startTime + duration
+    const kind = LONG_TASK
+    const span = new Span(`Longtask(${name})`, kind, { startTime })
```

Comment:
can `name` ever be undefined?

Reply:
it defaults to `unknown`, so not really.